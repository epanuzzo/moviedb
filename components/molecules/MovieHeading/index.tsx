import React from "react";

export type MovieHeadingProps = {
  title: string;
  tagline: string;
  description: string;
};

/**
 * MovieHeading component displays the main heading, tagline, and description of a movie.
 *
 * @component
 * @param {Object} props - The properties object.
 * @param {string} props.title - The title of the movie.
 * @param {string} props.tagline - The tagline of the movie.
 * @param {string} props.description - The description of the movie.
 * @returns {JSX.Element} The rendered MovieHeading component.
 */
const MovieHeading: React.FC<MovieHeadingProps> = ({
  title,
  tagline,
  description,
}) => {
  return (
    <div className="mt-6">
      <h1 className="text-4xl font-bold">{title}</h1>
      <p className="text-gray-400 italic">{tagline}</p>
      <p className="mt-4 text-lg">{description}</p>
    </div>
  );
};

export default MovieHeading;
