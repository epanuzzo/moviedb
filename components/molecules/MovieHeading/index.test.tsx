import React from "react";
import renderer from "react-test-renderer";
import MovieHeading from "./index";

describe("MovieHeading", () => {
  it("renders component", () => {
    const tree = renderer
      .create(
        <MovieHeading
          title="Movie Title"
          tagline="Tagline"
          description="Description"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
