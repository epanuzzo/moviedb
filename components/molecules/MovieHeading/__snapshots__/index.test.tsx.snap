// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieHeading renders component 1`] = `
<div
  className="mt-6"
>
  <h1
    className="text-4xl font-bold"
  >
    Movie Title
  </h1>
  <p
    className="text-gray-400 italic"
  >
    Tagline
  </p>
  <p
    className="mt-4 text-lg"
  >
    Description
  </p>
</div>
`;
